cdflib/cdfread.py

```diff
@@ -1,2254 +1,2254 @@
-        record_variance_bool = (flag_bits[31]=='1') 
-        pad_bool = (flag_bits[30]=='1')
-        compression_bool = (flag_bits[29]=='1') 
-        
-        sparse = int.from_bytes(vdr[40:44],'big', signed=True)
-        num_elements = int.from_bytes(vdr[56:60],'big', signed=True)
-        var_num = int.from_bytes(vdr[60:64],'big', signed=True)
-        CPRorSPRoffset = int.from_bytes(vdr[64:72],'big', signed=True)
-        blocking_factor = int.from_bytes(vdr[72:76],'big', signed=True)
-        name = str(vdr[76:332].decode('utf-8'))
-        name = name.replace('\x00', '')
-        
-        zdim_sizes = []
-        dim_sizes = []
-        dim_varys = []
-        if (section_type == 8):
-            #zvariable
-            num_dims = int.from_bytes(vdr[332:336],'big', signed=True)
-            for x in range(0, num_dims):
-                ioff = 336 + 4 * x
-                zdim_sizes.append(int.from_bytes(vdr[ioff:ioff+4],'big',
-                                                 signed=True))
-            coff = 336 + 4 * num_dims
-            for x in range(0, num_dims):
-                dim_varys.append(int.from_bytes(vdr[coff+4*x:coff+4*x+4],
-                                                'big', signed=True))
-            adj = 0
-            #Check for "False" dimensions, and delete them
-            for x in range(0, num_dims):
-                y = num_dims - x - 1
-                if (dim_varys[y]==0):
-                    del zdim_sizes[y]
-                    del dim_varys[y]
-                    adj = adj + 1
-            num_dims = num_dims - adj
-            coff = 336 + 8 * num_dims
-        else:
-            #rvariable
-            for x in range(0, self._rvariables_num_dims):
-                ioff = 332 + 4 * x
-                dim_varys.append(int.from_bytes(vdr[ioff:ioff+4],'big',
-                                                signed=True))
-            for x in range(0, self._rvariables_num_dims):
-                if (dim_varys[x]!=0):
-                    dim_sizes.append(self._rvariables_dim_sizes[x])
-            num_dims = len(dim_sizes)
-            coff = 332 + 4 * self._rvariables_num_dims
-        #Only set if pad value is in the flags
-        if pad_bool:
-            byte_stream = vdr[coff:]
-            pad = self._read_data(byte_stream, data_type, 1, num_elements)
- 
-        return_dict = {}
-        return_dict['data_type'] = data_type
-        return_dict['section_type'] = section_type
-        return_dict['next_vdr_location'] = next_vdr
-        return_dict['variable_number'] = var_num
-        return_dict['head_vxr'] = head_vxr
-        return_dict['last_vxr'] = last_vxr
-        return_dict['max_records'] = max_rec
-        return_dict['name'] = name
-        return_dict['num_dims'] = num_dims
-        if (section_type == 8):
-            return_dict['dim_sizes'] = zdim_sizes
-        else:
-            return_dict['dim_sizes'] = dim_sizes
-        if (pad_bool):
-            return_dict['p
```

Comment:
When using hexlify, it looks like the number stays type "bytes".  So then in the next line, it is comparing bytes to strings